### ISSUE TITLE
How to make loss layer for multi-label classification?

### ISSUE THREAD

**Issue by nitish11** [fchollet/keras#761]
Hi,

I am working on a multi-label image classification using keras.

Something similar to this paper, http://arxiv.org/pdf/1312.6082v4.pdf

I am trying to implement the below architecture : 

![multi-digit-number-recognition](https://cloud.githubusercontent.com/assets/2395767/10228512/c53c73c8-6890-11e5-90f4-65daec69c723.png)

How to implement the final multiple softmax layers in Keras?? Is there any existing example which can be referred for the same??

**Comment by alvarouc** [comment]
This has been addressed in another issue, just put sigmoids on your output layer and binary_crossentropy on your cost function. It works good for me.

**Comment by nitish11** [comment]
@alvarouc : Thanks for the reply. Kindly share the link of that issue.

**Comment by alvarouc** [comment]
Issue #741

**Comment by ritchieng** [comment]
@nitish11 Did you manage to figure a way out? I can't seem to figure out how we can have the independent fully connected layers at the end using Keras. They have merge but no "branch out" if you get what I mean.

**Comment by talbaumel** [comment]
Any idea why do I get much much better results when using categorical_crossentropy instead of binary_crossentropy when my target vectors are zeros and ones? 

binary_crossentropy shouldn't work well if the output is not a distribution

**Comment by nitish11** [comment]
@ritchieng : You can refer a link below. 
https://suraj-deshmukh.github.io/Multi-Label-Image-Classification/

@talbaumel : I haven't tried benchmarking any experiment like this.

**Comment by talbaumel** [comment]
You shouldn't it's wrong, but for some strange reason it gives me better results

**Comment by mrgloom** [comment]
@talbaumel I have opposite behaviour using MNIST dataset I have better accuracy when I use sigmoid+binary_crossentropy vs softmax+categorical_crossentropy but it seems only because I'm not leave just one output class.

Here is valid code for test:

	y_pred= model.predict(X_test)
	y_pred_= np.zeros((y_test.shape[0],),np.uint8)
	y_true_= np.zeros((y_test.shape[0],),np.uint8)
	for i in range(y_pred.shape[0]):
		y_pred_[i]= y_pred[i].argmax()
		y_true_[i]= y_test[i].argmax()
	
	print 'y_true_.shape',y_true_.shape
	print 'y_pred_.shape',y_pred_.shape
	print "acc:",accuracy_score(y_true_, y_pred_)
